spec/lib/sign_in/idme/service_spec.rb

```diff
@@ -213,6 +213,44 @@
       end
     end
 
+    context 'when selecting ID.me public certificates' do
+      let(:idme_keys) { JSON.parse(JSON.parse(File.read('spec/fixtures/sign_in/idme_public_keys.json'))) }
+      let(:stored_idme_certs) { idme_keys.map { |key| OpenSSL::PKey::RSA.new key } }
+      let(:config) { subject.send(:config) }
+
+      around { |example| VCR.use_cassette('identity/idme_200_responses', &example) }
+
+      it 'calls the ID.me configuration class jwt_decode_public_key method' do
+        expect_any_instance_of(SignIn::Idme::Configuration).to receive(:jwt_decode_public_key)
+          .and_return(stored_idme_certs)
+        subject.user_info(token)
+      end
+
+      context 'when the service config @idme_public_keys instance variable exists' do
+        before { config.instance_variable_set(:@idme_public_keys, stored_idme_certs) }
+
+        it 'does not call the IdmePublicKeyFetcher for updated certificates' do
+          expect_any_instance_of(SignIn::IdmePublicKeyFetcher).not_to receive(:perform)
+          subject.user_info(token)
+        end
+      end
+
+      context 'when the service config @idme_public_keys instance variable does not exist' do
+        before { config.instance_variable_set(:@idme_public_keys, nil) }
+
+        it 'calls the IdmePublicKeyFetcher for updated certificates' do
+          expect_any_instance_of(SignIn::IdmePublicKeyFetcher).to receive(:perform).and_return(stored_idme_certs)
+          subject.user_info(token)
+        end
+
+        it 'caches the response the service config @idme_public_keys instance variable' do
```

Comment:
We can prob remove this test, the one above that makes sure the fetcher isn't called on subsequent calls should be enough